projects/packages/search/src/class-options.php

```diff
@@ -48,6 +48,7 @@ class Options {
 	const OVERLAY_TRIGGER_IMMEDIATE = 'immediate';
 	const OVERLAY_TRIGGER_RESULTS   = 'results';
 	const OVERLAY_TRIGGER_SUBMIT    = 'submit';
+	const DEFAULT_OVERLAY_TRIGGER   = 'submit';
```

Comment:
To preserve the existing behaviour for older Jetpack versions, we could make this a method called `get_default_overlay_trigger()`, and that returns 'immediate' or 'results' for JP < 11.2 or 'submit' otherwise.